backend/src/org/commcare/resources/ResourceManager.java

```diff
@@ -232,7 +232,7 @@ public void upgrade()
                 //everything here? Locale files are registered elsewhere, and we
                 //can't guarantee we're the only thing in there, so we can't
                 //straight up clear it...
-                platform.initialize(masterTable);
+                platform.initialize(masterTable, true);
```

Comment:
This step isn't really guaranteed to happen. IE: If the upgrade is successful and the app crashes in platform.clearAppState(), we won't process the new fixture.